SwiftUICatalog/Views/Sections/Gestures/GesturesView.swift

```diff
@@ -37,8 +37,159 @@ import SwiftUI
 ///
 
 struct GesturesView: View {
+    
+    @GestureState var isDetectingLongPress = false
+    @State var totalNumberOfTaps = 0
+    @State var customText = "No Tap Detected yet!"
+    
     var body: some View {
-        Text("Gestures in SwiftUI")
+        
+        //MARK: Gestures
+        let tap = TapGesture()
+            .onEnded{ _ in
+                self.customText = "Tap Detected!"
+                print("View tapped!")
+            }
+        
+        let lognPress = LongPressGesture(minimumDuration: 2)
+            /**
+             - Update Transient UI State
+             To update a view as a gesture changes, add a GestureState property to your view and update it in the updating(_:body:) callback. SwiftUI invokes the updating callback as soon as it recognizes the gesture and whenever the value of the gesture changes. For example, SwiftUI invokes the updating callback as soon as a magnification gesture begins and then again whenever the magnification value changes. SwiftUI doesn’t invoke the updating callback when the user ends or cancels a gesture. Instead, the gesture state property automatically resets its state back to its initial value.
+             */
+            .updating($isDetectingLongPress) { currentState, gestureState, transaction in
+                gestureState = currentState
+            }
+            /**
+             - Update Permanent State During a Gesture
+             To track changes to a gesture that shouldn’t reset once the gesture ends, use the onChanged(_:) callback. For example, to count the number of times your app recognizes a long press, add an onChanged(_:) callback and increment a counter.
+             */
+            .onChanged {_ in
+                self.totalNumberOfTaps += 1
+            }
+            /**
+             - Update Permanent State When a Gesture Ends
```

Comment:
ah nice to include some documentation comments block, will take the idea for other future additions I do.